lib/codegen/meta/base/settings.py

```diff
@@ -91,6 +91,12 @@
         """,
         default=True)
 
+enable_nan_canonicalization = BoolSetting(
+        """
+        Enable NaN canonicalization
+        """,
+        default=True)
```

Comment:
I think the default should be `False`, since wasm doesn't require this behavior.